app/src/test/java/org/simple/clinic/drugs/selectionv2/entry/CustomPrescriptionEntryControllerv2Test.kt

```diff
@@ -151,4 +151,20 @@ class CustomPrescriptionEntryControllerv2Test {
     verify(sheet).setMedicineName(prescription.name)
     verify(sheet).setDosage(prescription.dosage)
   }
+
+  @Test
+  fun `when sheet is opened in edit mode and save is clicked after making changes, then the prescription should be updated`(){
+    val prescribedDrug = PatientMocker.prescription(uuid = prescriptionUuid, name = "Atnlol", dosage = "20mg")
+
+    whenever(prescriptionRepository.prescription(prescriptionUuid)).thenReturn(Observable.just(prescribedDrug))
+    whenever(prescriptionRepository.updatePrescription(any())).thenReturn(Completable.complete())
+
+    uiEvents.onNext(CustomPrescriptionSheetCreated(OpenAs.Update(prescriptionUuid)))
+    uiEvents.onNext(CustomPrescriptionDrugNameTextChanged("Atenolol"))
+    uiEvents.onNext(CustomPrescriptionDrugDosageTextChanged("5mg"))
+    uiEvents.onNext(SaveCustomPrescriptionClicked())
+
+    verify(prescriptionRepository).updatePrescription(prescribedDrug.copy(name = "Atenolol", dosage = "5mg"))
+    verify(sheet).finish()
```

Comment:
Let's also add assertions for checking the inverse functionality? That is `prescriptionRepository.savePrescription()` never gets called here and vice versa.

Reply:
Duplicate comment.